lib/disability_compensation/providers/document_upload/evss_supplemental_document_upload_provider.rb

```diff
@@ -47,24 +57,66 @@ def validate_upload_document(evss_claim_document)
   #
   # @return [Faraday::Response] The EVSS::DocumentsService API calls are implemented with Faraday
   def submit_upload_document(evss_claim_document, file_body)
+    log_upload_attempt
+
     client = EVSS::DocumentsService.new(@form526_submission.auth_headers)
     client.upload(file_body, evss_claim_document)
-  end
 
-  def log_upload_success(uploading_class_prefix)
-    StatsD.increment("#{uploading_class_prefix}.#{STATSD_PROVIDER_METRIC}.#{STATSD_SUCCESS_METRIC}")
+    # EVSS::DocumentsService uploads throw a EVSS::ErrorMiddleware::EVSSError if they fail
+    # If no exception is raised, log a success response
+    log_upload_success
+  rescue EVSS::ErrorMiddleware::EVSSError => e
```

Comment:
Note since the EVSS::DocumentsService has some custom error response logic that involves throwing exceptions (based on the test [here](https://github.com/department-of-veterans-affairs/vets-api/blob/benefits-disability-nb-refactor-supplemental-document-upload-providers/spec/lib/evss/documents_service_spec.rb#L49)) we don't have an easy way of capturing an actual upload response body here. But we want to log the failure event and re-raise the exception to preserve existing behavior.

I didn't dig too much into the EVSS exception handling or response format and I'm not sure we want to for this, @kylesoskin these failures are getting logged somewhere I assume, do you think it's sufficient for this migration to just log the event as I've done here? I also wasn't sure if we worried about logging PII from the EVSS responses. Did you have any insights/opinions here based on your historical knowledge?

Reply:
I think for EVSS, we just do our best to rescue and statsD/log whatever we can. I think this is probably good enough